test-runner/src/main/scala/org/typesafe/spark/mesos/framework/runners/MesosIntegrationTestRunner.scala

```diff
@@ -17,9 +17,8 @@ object MesosIntegrationTestRunner {
     printMsg("TestResults:")
     result.foreach(printMsg)
 
-    //TODO: Generate some JUnit style report for CI
-    if (result.exists(_.startsWith("TestFailed"))) {
-      System.exit(1)
+    if (result.exists(_.contains("FAILED"))) {
+      printMsg(Console.RED + "ERROR: One or more tests failed" + Console.RESET)
```

Comment:
You lost the `System.exit(1)`. Without it, sbt act link if everything when fine even if one test fails. It is better to return a failure if one or more test failed.